examples/module1/lesson1/validate-it/index.js

```diff
@@ -1,31 +1,29 @@
 function validator() {
   const input = document.getElementById('input');
-  const button = document.getElementById('button');
-  const button2 = document.getElementById('button2');
+  const validateButton = document.getElementById('validateButton');
+  const clearButton = document.getElementById('clearButton');
   const result = document.getElementById('result');
 
-  button.addEventListener('click', () => {
-    if (input.value) {
-      if (Number.isInteger(input.value)) {
-        if (
-          Number(input.value) > 0 &&
-          Number(input.value) < 100 &&
-          Number(input.value) % 2 === 0
-        ) {
-          result.innerHTML = 'Valid';
-        } else {
-          result.innerHTML = 'Invalid';
-        }
-        result.innerHTML = 'Valid';
-      } else {
-        result.innerHTML = 'Invalid';
-      }
-    } else {
-      result.innerHTML = 'Invalid';
+  validateButton.addEventListener('click', () => {
+    const MAX_VALUE = 100;
+    const MIN_VALUE = 0;
+    const inputValue = Number(input.value);
+
+    if(!Number.isInteger(inputValue)){
+      result.innerHTML = 'INVALID';
+      return;
+    }
+
+    if(inputValue > MIN_VALUE && inputValue < MAX_VALUE && inputValue % 2 === 0){
```

Comment:
tutaj stworzyłbym na zweryfikowanie każdego z tych warunków osobną funkcję np. `isHigherThan`, `isLowerThan` i `isEven` wtedy można by pomyśleć nad tym punktem, który był jednym z wymagań w zadaniu:

> - Jak sprawić, żeby dodawanie i usuwanie kolejnych reguł walidacji nie wpływało na każdorazową edycję funkcji do walidacji?

Reply:
samą logikę walidacyjną przeniósłbym także do osobnej funkcji, pamiętaj że możemy to przepisać na typescript, co może sporo ułatwić